Merge with smz3 sprite

Serthow opened the pull request:
Due to Weirdness with github, creadted an extra branch to clone unstable and open a pull request from

Serthow committed:
Merge with smz3 sprite

Due to Weirdness with github, creadted an extra branch to clone unstable and open a pull request from

Serthow committed:
Tab to space on all whitespace

Serthow committed:
Tab to Space on SMZ3_sprite

Serthow committed:
Remove smz3_sp[rite

Removed SMZ3_sprite.as and moved all of its methods into GameState and init

Edited the call to the method in preframe accordingly

JamesDunne commented on alttpo/GameState.as:
```diff
@@ -1112,9 +1137,256 @@ class GameState {
+        case 0x82: return 5;
+        case 0xa4: return 7;
+        case 0xa5: return 7;
+        default: return 0;
```
Are you missing 0x19 and 0x1a poses for spin jump here?

Serthow replied:
for whatever reason those poses appear fine with the default protocol, so i didn't need to list them

JamesDunne replied:
oh really? i'm guessing it depends on what equipment you have. i know i saw 19 and 1a poses on a clean startup without screw attack

Serthow committed:
update to sm_event

Changing the sm_event protocol to incorporate small keys

only GameState got updated and caused a mismatch in the size of the array

Serthow committed:
Bug fixes

space jump withoug screw attack is it's own pose and wasn't caught

the map window has been updated to reflect thew split between the room position and smaus position in the room

the game no longer updates the sprite data while local player is in the menu, which was causing an invalid sprite to show

Serthow commented on alttpo/GameState.as:
```diff
@@ -1112,9 +1137,256 @@ class GameState {
+    uint32 transfer1;
+    uint16 len1;
+    uint16 len2;
+    if (sm_pose == 0x1a || sm_pose == 0x19 || sm_pose == 0x81 || sm_pose == 0x82 || sm_pose == 0x1b || sm_pose == 0x1c){
```
line 1194: added the space jump poses to the check
